lib/parsers/csv_layout/csv_parser.rb

```diff
@@ -90,12 +103,8 @@ def validate_location_duplication
       end
 
       def duplicated(sym)
-        all_elems = layout.map { |obj| obj[sym] }
-        all_elems.select do |element|
-          (!element.nil?) && (all_elems.count(element) > 1)
-        end.uniq.compact
+        layout.pluck(sym).compact.uniq! || []
```

Comment:
uniq! does not return the list, but nil, so it will be always [] I think

Reply:
Uniq! only returns nil if there are no duplicates, otherwise it returns an array of the duplicates.